bin/resize_partition.ml

```diff
@@ -0,0 +1,86 @@
+open Cmdliner
+
+let read_mbr mbr =
+  let ic = open_in_bin mbr in
+  let buf = Bytes.create Mbr.sizeof in
+  let () = really_input ic buf 0 Mbr.sizeof in
+  close_in ic;
+  match Mbr.unmarshal (Cstruct.of_bytes buf) with
+  | Ok mbr -> (mbr, Mbr.sizeof)
+  | Error msg ->
+      Printf.printf "Failed to read MBR from %s: %s\n" mbr msg;
+      exit 1
+
+let get_partition_info mbr partition_number =
+  List.nth mbr.Mbr.partitions (partition_number - 1)
+
+let calculate_partition_info partition =
+  (* FIXME: Use Int32.unsigned_to_int *)
+  let start_sector =
+    Int32.to_int partition.Mbr.Partition.first_absolute_sector_lba
+  in
+  let sector_size = 512 in
+  (start_sector, sector_size)
+
+let make_new_partition partition start_sector sector_size new_size =
+  let new_num_sectors = new_size / sector_size in
```

Comment:
This is a bit "dangerous". If the user asks for 1000 bytes, then the partition will end up being just a single sector (512 bytes).

Alternatives are to round up, error when the size is not sector aligned or use number of sectors as the size. The first has the issue that it does not exactly what the user asked for (even if it doesn't make sense). The second introduces another check in the code. The third leads to simpler code, but may be less user friendly as the user has to do bytes-to-sectors calculations themselves.

Reply:
Thank you for pointing this out. I will look into implementing the second option.

Reply:
But this will mean partitions can be resized only to multiples of 512